src/notation/view/widgets/scoreproperties.cpp

```diff
@@ -248,31 +249,31 @@ bool ScorePropertiesDialog::save()
         QLayoutItem* tagItem   = scrollAreaLayout->itemAtPosition(i, 0);
         QLayoutItem* valueItem = scrollAreaLayout->itemAtPosition(i, 1);
         if (tagItem && valueItem) {
-            QLineEdit* tag   = static_cast<QLineEdit*>(tagItem->widget());
-            QLineEdit* value = static_cast<QLineEdit*>(valueItem->widget());
 
-            QString tagText = tag->text();
+            ScorePropertiesItem item(tagItem->widget(), valueItem->widget());
```

Comment:
This class is a bit redundant. You can just add missing nullptr checks where the crash occurs

Reply:
Is that why the test runs have all failed except on windows? Have I added the file incorrectly?

I split it out into a new class to a) fullfill the single responsibility principle of SOLID and b) it made it much easier to unit test, but if it creates other issues then yes of course just doing the cast check in the score properties page will work. 

I’ll have a think how I can TDD that…

Reply:
Ps I think you are correct btw - I’ll go and simplify.

Reply:
These principles are good but in this case, your solution overcomplicates the dialog implementation. In addition, the dialog will be rewritten using the qml / c++ pair in the future (or completely replaced with something else). Anyway, thanks for your work!

Reply:
Simple fix as per above on way....